easycv/dataset.py

```diff
@@ -0,0 +1,69 @@
+from easycv import List
+from easycv.collection import Collection, auto_compute
+from easycv.errors.dataset import InvalidClass, NoClassesGiven
+
+import os
+
+
+class Dataset(Collection):
+    def __init__(self, path, pipeline=None, recursive=False):
+        super().__init__(pending=pipeline)
+        self.path = path
+        self.dataset = {}
+        self._size = -1
+        self.setup(path, recursive)
+
+    def setup(self, path, recursive):
+        classes = next(os.walk(path))[1]
+        if not classes:
+            raise NoClassesGiven()
+        for clas in classes:
+            self.dataset[clas] = List(os.path.join(path, clas), lazy=True, recursive=recursive)
+
+    @property
+    def classes(self):
+        return list(self.dataset.keys())
+
+    @auto_compute
+    def get(self, key):
+        if key not in self.dataset:
+            raise InvalidClass(key)
+        return self.dataset[key]
+
+    def size(self):
+        if self._size == -1:
+            self._size = sum([len(self.dataset[clas]) for clas in self.dataset])
+        return self._size
+
```

Comment:
Missing auto compute

Reply:
Actually you dont need to compute to measure the len. Implement a lazy len to prevent this.